cmake/mcuboot.cmake

```
@@ -50,6 +50,30 @@ function(zephyr_mcuboot_tasks)
     message(FATAL_ERROR "Can't sign images for MCUboot: Neither CONFIG_BUILD_OUTPUT_BIN nor CONFIG_BUILD_OUTPUT_HEX is enabled, so there's nothing to sign.")
   endif()
 
+  # Find encryption key file, if specified
+  set(keyfile_enc "${CONFIG_MCUBOOT_ENCRYPTION_KEY_FILE}")
+  if(NOT "${keyfile_enc}" STREQUAL "")
+
+    if(NOT IS_ABSOLUTE "${keyfile_enc}")
+      # Relative paths are relative to 'west topdir'.
+      set(keyfile_enc "${WEST_TOPDIR}/${keyfile_enc}")
+      set(keyfile_enc_relative TRUE)
+    else()
+      set(keyfile_enc_relative FALSE)
+    endif()
+
+    if(NOT EXISTS "${keyfile_enc}")
+      if(keyfile_enc_relative)
+        set(relative_msg " Note: relative paths are relative to the west workspace topdir \"${WEST_TOPDIR}\".")
+      else()
+        set(relative_msg "")
+      endif()
+      message(FATAL_ERROR "Can't encrypt images for MCUboot: CONFIG_MCUBOOT_ENCRYPTION_KEY_FILE=\"${CONFIG_MCUBOOT_ENCRYPTION_KEY_FILE}\" not found.${relative_msg}")
+    elseif(NOT (CONFIG_BUILD_OUTPUT_BIN OR CONFIG_BUILD_OUTPUT_HEX))
+      message(FATAL_ERROR "Can't encrypt images for MCUboot: Neither CONFIG_BUILD_OUTPUT_BIN nor CONFIG_BUILD_OUTPUT_HEX is enabled, so there's nothing to encrypt.")
+    endif()
+  endif()
```

This code block is almost an identical copy of this:
https://github.com/zephyrproject-rtos/zephyr/blob/93e674e37382dffe084d03fc2c51d85d898c2e7f/cmake/mcuboot.cmake#L34-L51

Please combine those two blocks into a single code block, for example like:
```suggestion
set(keyfile_enc "${CONFIG_MCUBOOT_ENCRYPTION_KEY_FILE}")
set(keyfile "${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}")

foreach(file keyfile keyfile_enc)
  if(NOT "${${file}}" STREQUAL "")
    if(NOT IS_ABSOLUTE "${${file}}")
      # Relative paths are relative to 'west topdir'.
      set(${file} "${WEST_TOPDIR}/${${file}}")
      set(${file}_relative TRUE)
    else()
      set(${file}_relative FALSE)
    endif()
 
    if(NOT EXISTS "${${file}}")
       ....
   
endforeach()
```

Adjust error messages accordingly.